### PR TITLE
chore: revamp experiment and trial pages header [DET-5406]

### PR DESCRIPTION
**Pull request by apizzini**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/styles/icons.scss [hunk 1]
```diff
@@ -4,8 +4,8 @@
   font-style: normal;
   font-weight: normal;
   src:
-    url('fonts/determined-ai.woff2?wo5opj') format('woff2'),
-    url('fonts/determined-ai.woff?wo5opj') format('woff');
+    url('fonts/determined-ai.woff2?wo6opj') format('woff2'),
```
Comment (apizzini) [inline webui/react/src/styles/icons.scss:7]: @hkang1 not sure if there are rules to change this hash?

File: webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  useEffect(() => {
+    setIsRunningArchive(false);
+    setIsRunningUnarchive(false);
+  }, [ experiment.archived ]);
+
+  const headerOptions = useMemo<Option[]>(() => {
+    const options: Option[] = [
+      {
+        icon: <Icon name="tensorboard" size="small" />,
```
Comment (apizzini) [inline webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx:36]: @hkang1 i think the fork icon is missing.

File: webui/react/src/components/PageHeaderFoldable.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import { Button, Dropdown, Menu } from 'antd';
+import React, { CSSProperties, useState } from 'react';
+
+import Icon from 'components/Icon';
+
+import css from './PageHeaderFoldable.module.scss';
```
Comment (apizzini) [inline webui/react/src/components/PageHeaderFoldable.tsx:6]: hover states needs to be revisited, @hkang1 have you already ideas about this?

File: webui/react/src/pages/ExperimentDetails/Header/ExperimentHeaderLeft.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import { ExperimentBase } from 'types';
+
+import css from './ExperimentHeaderLeft.module.scss';
+import ExperimentState from './ExperimentHeaderState';
+
+interface Props {
+  experiment: ExperimentBase;
+}
+
+const ExperimentHeaderLeft: React.FC<Props> = ({ experiment }: Props) => {
```
Comment (apizzini) [inline webui/react/src/pages/ExperimentDetails/Header/ExperimentHeaderLeft.tsx:12]: @hkang1 we might consider including this directly in `webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx`

File: webui/react/src/pages/ExperimentDetails/Header/ExperimentHeaderFoldableContent.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import { ExperimentBase } from 'types';
+
+import css from './ExperimentHeaderFoldableContent.module.scss';
+
+interface Props {
+  experiment: ExperimentBase;
+  fetchExperiment: () => void;
+}
+
+const ExperimentHeaderFoldableContent: React.FC<Props> = (
```
Comment (apizzini) [inline webui/react/src/pages/ExperimentDetails/Header/ExperimentHeaderFoldableContent.tsx:14]: @hkang1 we might consider including this directly in `webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx`

File: webui/react/src/pages/ExperimentDetails/Header/ExperimentHeaderFoldableContent.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  { experiment, fetchExperiment }: Props,
+) => {
+  const experimentTags = useExperimentTags(fetchExperiment);
+
+  return (
+    <>
+      <div className={css.description}>
+        Placeholder for experiment description.
```
Comment (apizzini) [inline webui/react/src/pages/ExperimentDetails/Header/ExperimentHeaderFoldableContent.tsx:22]: @hkang1 this is of course a todo.

File: webui/react/src/components/PageHeaderFoldable.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+interface Props {
+  foldableContent?: React.ReactNode,
+  leftContent: React.ReactNode,
+  options?: Option[],
+  style?: CSSProperties;
+}
+
+// const wrapWithTooltip = (text: string|undefined, component: React.ReactNode): React.ReactNode => {
```
Comment (apizzini) [inline webui/react/src/components/PageHeaderFoldable.tsx:24]: we have 1 case (trial with no checkpoint) where we would need to show a tooltip on buttons.

We have 2 issues with this:
- tooltip don't work too fine on mobile
- `<Menu>` doesn't work fine if `<Menu.Item>` are wrapped in `<Tooltip>`

Possible alternative solutions are to hide the btn or to make it clickable (not disabled) and open a modal to show the error msg. Other ideas?

File: webui/react/src/components/PageHeaderFoldable.tsx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+interface Props {
+  foldableContent?: React.ReactNode,
+  leftContent: React.ReactNode,
+  options?: Option[],
+  style?: CSSProperties;
+}
+
+// const wrapWithTooltip =
```
Comment (apizzini) [inline webui/react/src/components/PageHeaderFoldable.tsx:24]: we have 1 case (trial with no checkpoint) where we would need to show a tooltip on buttons.

We have 2 issues with this:

tooltip don't work too fine on mobile
<Menu> doesn't work fine if <Menu.Item> are wrapped in <Tooltip>

Possible alternative solutions are to hide the btn or to make it clickable (not disabled) and open a modal to show the error msg. Other ideas?

File: webui/react/src/styles/icons.scss (hunk 1 above)
Comment (hkang1) [reply webui/react/src/styles/icons.scss:7]: we change the icon font so infrequently I just use the icomoon generated hash to replace it directly

File: webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx (hunk 2 above)
Comment (hkang1) [reply webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx:36]: ah I'll get this to you

File: webui/react/src/components/PageHeaderFoldable.tsx (hunk 8 above)
Comment (hkang1) [reply webui/react/src/components/PageHeaderFoldable.tsx:24]: I think if the tooltip is inside the `Menu.Item` but around the label text it should work? Not 100% sure, I believe `NavigationSideBar` has a working example of it?

File: webui/react/src/components/PageHeaderFoldable.tsx (hunk 3 above)
Comment (hkang1) [reply webui/react/src/components/PageHeaderFoldable.tsx:6]: I'll need to check out this branch and try it out to get back to you on this point.

File: webui/react/src/components/PageHeaderFoldable.tsx (hunk 8 above)
Comment (apizzini) [reply webui/react/src/components/PageHeaderFoldable.tsx:24]: This works but is not so solid.

The tooltip is triggered only when focusing the text inside the Menu.Item, not the whole Menu.Item.

File: webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+) => {
+  const [ isRunningTensorboard, setIsRunningTensorboard ] = useState<boolean>(false);
+
+  const headerOptions = useMemo<Option[]>(() => {
+    const options: Option[] = [];
+
+    if (trial.bestAvailableCheckpoint !== undefined) {
+      options.push({
+        key: Action.Continue,
+        label: 'Continue Trial',
```
Comment (apizzini) [inline webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx:28]: do we have an icon for this?

File: webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+        tooltip: 'No checkpoints found. Cannot continue trial',
+      });
+    }
+
+    if (!trialWillNeverHaveData(trial)) {
+      options.push({
+        isLoading: isRunningTensorboard,
+        key: Action.Tensorboard,
+        label: 'TensorBoard',
```
Comment (apizzini) [inline webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx:43]: for this i guess we want to use the same icon of experiment header?

File: webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx (hunk 9 above)
Comment (hkang1) [reply webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx:28]: It's a different variation of forking, so maybe just use the `fork` icon this too

**Review by hkang1 — APPROVED** [review]
LGTM, this is looking very nice!

File: webui/react/src/services/decoder.ts [hunk 11]
```diff
@@ -286,10 +288,10 @@ const mapV1Experiment = (
 ): types.ExperimentItem => {
   return {
     archived: data.archived,
+    description: data.description,
     endTime: data.endTime as unknown as string,
     id: data.id,
     labels: data.labels || [],
-    // capture description for use in the UI.
```
Comment (apizzini) [inline webui/react/src/services/decoder.ts:292]: @hamidzr can you please check if the changes done in those 2 files of this MR are ok?
- webui/react/src/services/decoder.ts
- webui/react/src/types.ts

**Pull request merged**